pipeline/assets/traffic_incidents.py

```diff
@@ -82,8 +88,18 @@ def roadworks_geojson() -> dict:
     io_manager_key='pg_gpd_io_manager',
     auto_materialize_policy=AutoMaterializePolicy.eager(),
 )
-def roadworks(roadworks_geojson) -> pd.DataFrame:
+def roadworks(roadworks_geojson: dict[str, Any]) -> pd.DataFrame:
     """
     Imports the roadworks into PostGIS, from where they can be accessed e.g. via WMS/WFS.
+    Note: only roadworks with geom_type are imported!
     """
-    return gpd.GeoDataFrame.from_features(roadworks_geojson['features'], crs='epsg:4326')
+    roadworks_gdf = gpd.GeoDataFrame.from_features(roadworks_geojson['features'], crs='epsg:4326')
+    roadworks_not_linestrings = roadworks_gdf[roadworks_gdf.geom_type != 'LineString']
+    if len(roadworks_not_linestrings) > 0:
+        logger.warn(
+            f'''Ignored {len(roadworks_not_linestrings)} which had
+            geom_type!=LineString, e.g. with id="{roadworks_not_linestrings["id"].iloc[0]}'''
+        )
+        roadworks_only_linestrings_gdf = roadworks_gdf[roadworks_gdf.geom_type == 'LineString']
+        return roadworks_only_linestrings_gdf.set_index('id')
+    return roadworks_gdf.set_index('id')
```

Comment:
`set_index` returns changed labels. Is that the same as before?

Reply:
It returns a DataFrame with changed labels and a named index. This index is downstream used to create the primary key column(s).

Reply:
Ok, fine :)